File: frontend/ClickAndDrive/src/app/services/report.service.ts
// Report service
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { ReportRequest } from "./models/report-request";
import { ReportResponse } from "./models/report-response";
import { Observable } from "rxjs";


@Injectable({
    providedIn: 'root'
})
export class ReportService {
    private apiUrl = 'http://localhost:8080/api/reports';

    constructor(private http: HttpClient) {}

    // Generate a report function
    generateReport(request: ReportRequest): Observable<ReportResponse> {
        return this.http.post<ReportResponse>(`${this.apiUrl}/generate`, request);
    }
}

File: mobile-version/Navbar-shortened/app/src/main/java/com/example/clickanddrive/clients/services/ReportService.java
package com.example.clickanddrive.clients.services;

import com.example.clickanddrive.dtosample.requests.ReportRequest;
import com.example.clickanddrive.dtosample.responses.ReportResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface ReportService {

    @POST("reports/generate")
    Call<ReportResponse> generateReport(@Body ReportRequest request);
}
